tests/laplace.hpp

```diff
@@ -75,13 +84,36 @@ Laplace<dim, VectorType>::Laplace(MPI_Comm const &comm, unsigned int fe_degree)
 }
 
 template <int dim, typename VectorType>
-void Laplace<dim, VectorType>::setup_system(int num_refinements)
+void Laplace<dim, VectorType>::setup_system(
+    boost::property_tree::ptree const ptree)
 {
-  dealii::GridGenerator::hyper_cube(_triangulation);
-  _triangulation.refine_global(num_refinements);
+  std::string const mesh = ptree.get("mesh", "hyper_cube");
+  if (mesh == "hyper_ball")
+  {
+    dealii::GridGenerator::hyper_ball(_triangulation);
+    _triangulation.set_all_manifold_ids_on_boundary(0);
+    _triangulation.set_manifold(0, _spherical_manifold);
+  }
+  else
+    dealii::GridGenerator::hyper_cube(_triangulation);
+
+  _triangulation.refine_global(ptree.get("n_refinements", 3));
+
+  if (ptree.get("distort_random", false))
+    dealii::GridTools::distort_random(0.2, _triangulation);
 
   _dof_handler.distribute_dofs(_fe);
 
+  std::string const renumbering = ptree.get("renumbering", "None");
```

Comment:
Please change "renumbering" to more commonly used "reordering" and add it to the input deck.